### PR TITLE
[Grafana] Create Az.Dashboard module and use apiverison = 2022-08-01

### PR DESCRIPTION
**Pull request by lijinpei2008**
## Description
cmdlet: New/Get/Update/Remove-AzGrafana

Swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/dashboard/resource-manager/Microsoft.Dashboard/stable/2022-08-01/grafana.json

Design: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1236

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from the scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** change version of module in pull request

### CODE REVIEW

File: src/Dashboard/docs/New-AzGrafana.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+Create or update a workspace for Grafana resource.
+This API is idempotent, so user can either create a new grafana or update an existing grafana.
+
+## SYNTAX
+
+```
+New-AzGrafana -Name <String> -ResourceGroupName <String> [-SubscriptionId <String>] [-ApiKey <ApiKey>]
+ [-AutoGeneratedDomainNameLabelScope <AutoGeneratedDomainNameLabelScope>]
+ [-DeterministicOutboundIP <DeterministicOutboundIP>] [-IdentityType <ManagedServiceIdentityType>]
+ [-IdentityUserAssignedIdentity <Hashtable>] [-Location <String>] [-PublicNetworkAccess <PublicNetworkAccess>]
```
Comment (BethanyZhou) [inline src/Dashboard/docs/New-AzGrafana.md:20]: Please update Location as required.

Comment (lijinpei2008) [reply src/Dashboard/docs/New-AzGrafana.md:20]: Swagger Link : https://github.com/Azure/azure-rest-api-specs/blob/main/specification/dashboard/resource-manager/Microsoft.Dashboard/stable/2022-08-01/grafana.json#L719
The swagger format was not "required":true

Comment (BethanyZhou) [reply src/Dashboard/docs/New-AzGrafana.md:20]: I remember you said it's required in your manual test?

Comment (lijinpei2008) [reply src/Dashboard/docs/New-AzGrafana.md:20]: Yes, but in fact, in the example I provided, all of the parameters used are required.
I am now ready to mark these parameters "Mandatory" under the Custom folder

Comment (BethanyZhou) [reply src/Dashboard/docs/New-AzGrafana.md:20]: I can't see it's mandatory in the syntax of New-AzGrafana.md

Comment (lijinpei2008) [reply src/Dashboard/docs/New-AzGrafana.md]: Update some parameters and mark them as required

**Review by BethanyZhou — APPROVED** [review]

**Comment by BethanyZhou** [comment]
I'm moving current PR to main branch. Please raise a PR to main branch after this pipeline finished https://dev.azure.com/azure-sdk/public/_build/results?buildId=1733989&view=results

**Pull request merged**